pkg/utils/detector.go

```diff
@@ -573,23 +573,27 @@ func GetAnyApplicationFilePathExactMatch(root string, propsFiles []model.Applica
 func GenerateApplicationFileFromFilters(files []string, path string, suffix string, ctx *context.Context) []model.ApplicationFileInfo {
 	applicationFileInfos := []model.ApplicationFileInfo{}
 	for _, file := range files {
-		if strings.HasSuffix(file, suffix) {
-			cleanPath := filepath.Clean(file)
-			filename := filepath.Base(cleanPath)
-			tmpDir := strings.ReplaceAll(file, path, "")
-			dir := strings.ReplaceAll(tmpDir, filename, "")
-			appFileInfo := model.ApplicationFileInfo{
-				Context: ctx,
-				Root:    path,
-				Dir:     dir,
-				File:    filename,
-			}
-			applicationFileInfos = append(applicationFileInfos, appFileInfo)
-		}
+		if strings.HasSuffix(file, suffix) && !strings.HasSuffix(file, "_test.go"){
+			applicationFileInfos = append(applicationFileInfos, createAppFileInfo(file, path, ctx))
+		}	
 	}
 	return applicationFileInfos
 }
 
+func createAppFileInfo(file string, path string, ctx *context.Context) model.ApplicationFileInfo {
```

Comment:
Initially I thought it would be okay to leave out an explicit test case for this function because it was being handled in `TestGenerateApplicationFileFromFilters` but I added tests for it anyways to be complete.